go/test/endtoend/vreplication/helper_test.go

```diff
@@ -368,29 +368,33 @@ func waitForWorkflowState(t *testing.T, vc *VitessCluster, ksWorkflow string, wa
 		require.NoError(t, err, output)
 		done = true
 		state := ""
-		streams := gjson.Get(output, "workflows.0.shard_streams.*.streams")
-		streams.ForEach(func(streamId, stream gjson.Result) bool { // For each stream
-			info := stream.Map()
-			// We need to wait for all streams to have the desired state.
-			state = info["state"].String()
-			if state == wantState {
-				for i := 0; i < len(fieldEqualityChecks); i++ {
-					if kvparts := strings.Split(fieldEqualityChecks[i], "=="); len(kvparts) == 2 {
-						key := kvparts[0]
-						val := kvparts[1]
-						res := info[key].String()
-						if !strings.EqualFold(res, val) {
-							done = false
+		shardStreams := gjson.Get(output, "workflows.0.shard_streams")
+		// We need to wait for all streams in all shard streams to have the desired state.
+		shardStreams.ForEach(func(shardStreamId, shardStream gjson.Result) bool {
```

Comment:
Have you used https://gjson.dev with sample outputs to be sure that this is different? I thought that this would get all streams, but maybe not:
```
streams := gjson.Get(output, "workflows.0.shard_streams.*.streams")
streams.ForEach(func(streamId, stream gjson.Result) bool { // For each stream
```

Just a note that there's no need to backport this PR as the code you're modifying here is new from this recent PR: https://github.com/vitessio/vitess/pull/17441

Reply:
OK, I confirmed that `workflows.0.shard_streams.*.streams` only returns 1 stream. Thanks!

Reply:
Yeah, I had tested the json parsing locally and that is how I figured out the problem and the fix.  The `gjson` document is quite confusing with regard to wildcard filtering!